Author: Daniel Baumann <mail@daniel-baumann.ch>
Description:
 Adding alternative locations in for configuration files for the
 bootloader configuration.

diff -Naurp syslinux.orig/core/fs/iso9660/iso9660.c syslinux/core/fs/iso9660/iso9660.c
--- syslinux.orig/core/fs/iso9660/iso9660.c
+++ syslinux/core/fs/iso9660/iso9660.c
@@ -233,12 +233,14 @@ static int iso_load_config(void)
 	"/isolinux",
 	"/boot/syslinux", 
 	"/syslinux", 
+	"/boot",
 	"/",
 	NULL
     };
     static const char *filenames[] = {
 	"isolinux.cfg",
 	"syslinux.cfg",
+	"boot.cfg",
 	NULL
     };
 
diff -Naurp syslinux.orig/core/fs/lib/loadconfig.c syslinux/core/fs/lib/loadconfig.c
--- syslinux.orig/core/fs/lib/loadconfig.c
+++ syslinux/core/fs/lib/loadconfig.c
@@ -15,14 +15,18 @@ int generic_load_config(void)
 {
     static const char *search_directories[] = {
 	NULL,			/* CurrentDirName */
+	"/boot/extlinux",
+	"/extlinux",
 	"/boot/syslinux",
 	"/syslinux",
+	"/boot",
 	"/",
 	NULL
     };
     static const char *filenames[] = {
 	"extlinux.conf",
 	"syslinux.cfg",
+	"boot.cfg",
 	NULL
     };
 
